### DISCUSSION TITLE
Query param typed as optional enum does not render as dropdown menu in Swagger's doc

### DISCUSSION THREAD

**Discussion by adrienballsonos** [fastapi/fastapi#10911] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from enum import Enum
from typing import Union

from fastapi import FastAPI

app = FastAPI()


class Country(str, Enum):
    FIRST_COUNTRY = "firstCountry"
    SECOND_COUNTRY = "secondCountry"


@app.get("/users")
def find_users(country: Union[Country, None] = None):
    # filter users on country
    print(f"Provided country: {country}")
```

### Description

* Open the Swagger page and check the `country` query param UI
* I expected to have a dropdown menu with two options (`firstCountry`, `secondContry`)
* But I got a text field (see first screenshot)
* Same result if I replace `Union[Country, None]` by `Optional[Country]`. I also tested with `Country | None` by switching to python 3.10 but it didn't work either.

### Operating System

macOS

### FastAPI Version

0.105.0

### Pydantic Version

2.5.2

### Python Version

3.9.9

### Additional Context

<img width="508" alt="Screenshot 2024-01-09 at 09 49 38" src="https://github.com/tiangolo/fastapi/assets/61321801/9f353b2a-d8d1-4be3-919c-5e30f3112371">

Here is what I have when I remove the optionality of the query param:
<img width="542" alt="Screenshot 2024-01-09 at 09 50 52" src="https://github.com/tiangolo/fastapi/assets/61321801/fdff21d6-c0ad-49f2-bce9-c69bb41398fd">
This is what I expected to have, without the red `*` as the param is optional.

**Comment by Kludex** [comment, marked as answer]
For now, my recommendation is to do the following:

```py
from enum import Enum

from fastapi import FastAPI
from pydantic.json_schema import SkipJsonSchema

app = FastAPI()


class Country(str, Enum):
    FIRST_COUNTRY = 'firstCountry'
    SECOND_COUNTRY = 'secondCountry'


@app.get('/users')
def find_users(country: Country | SkipJsonSchema[None] = None):
    # filter users on country
    print(f'Provided country: {country}')
```

There are already multiple discussions about this. FastAPI will solve this at some point.

**Reply by adrienballsonos** [reply]
My bad, I didn't search hard enough for similar discussions. There is already this one for instance, covering this issue: https://github.com/tiangolo/fastapi/discussions/10316.

Thanks for your help.

**Reply by MasterGroosha** [reply]
Using FastAPI 0.111, I can just use `country: Country = None` and it works fine.
